modules/apps/commerce/commerce-shipping-engine-fixed-web/src/main/java/com/liferay/commerce/shipping/engine/fixed/web/internal/ByWeightCommerceShippingEngine.java

```diff
@@ -301,6 +299,20 @@ private List<CommerceShippingOption> _getCommerceShippingOptions(
 			new CommerceShippingOptionPriorityComparator());
 	}
 
+	private BigDecimal _getOrderShippableSubtotal(CommerceOrder commerceOrder) {
+		BigDecimal subtotal = BigDecimal.ZERO;
+		List<CommerceOrderItem> commerceOrderItems =
+			commerceOrder.getCommerceOrderItems();
+
+		for (CommerceOrderItem commerceOrderItem : commerceOrderItems) {
+			if (commerceOrderItem.isShippable()) {
+				subtotal = subtotal.add(commerceOrderItem.getFinalPrice());
```

Comment:
I'm not 100% sure you should use this or `getFinalPriceWithTaxAmountMoney()`

I think there is a config that says how to calculate price.

Reply:
I think taxes are applied on top of this, as before the change where we were using the order's subtotal.

Reply:
Talked offline with @luca-pellizzon , the solution is right!
Thank you!